micronota/bfillings/infernal.py

```diff
@@ -0,0 +1,239 @@
+
+    cmfetch is used to get a covariance model by name or accession
+    from a CM database.
+    This wrapper is tested for INFERNAL 1.1.1 (July 2014).
+    '''
+    pass
+
+
+class CMPress(CommandLineApplication):
+    '''cmpress application controller.
+
+    cmpress is used to format a CM database into a binary format for cmscan.
+    This wrapper is tested for INFERNAL 1.1.1 (July 2014).
+    '''
+    _suppress_stderr = False
+    _command = 'cmpress'
+    _parameters = {
+        # force overwrite
+        '-F': FlagParameter(Prefix='-', Name='F')}
+
+    def _accept_exit_status(self, exit_status):
+        return exit_status == 0
+
+
+def cmpress_cm(cm, force=False):
+    '''Compress the CM database.
+
+    Parameters
+    ----------
+    cm : str
+        The file path to CM database.
+    force : boolean
+        Whether to overwrite.'''
+
+    app = CMPress(InputHandler='_input_as_path')
+    if force is True:
+        app.Parameters['-F'].on()
+    res = app(cm)
+    return res
+
+
+def cmscan_fasta(cm, in_fp, out_fp, evalue=0.01, cores=0, params=None):
+    '''Scan fasta file against a covariance model database.
+
+    Parameters
+    ----------
+    cm : str
+    in_fp : str
+        Input fasta file.
+    out_fp : str
+        Output file path of target hits table.
+    cores : int
+        Number of CPU cores. Default to zero, i.e. running in serial-only mode.
+    evalue : float
+        Default to 0.01. Threshold E-value.
+    params : dict
+        Other command line parameters for cmscan. key is the option
+        (e.g. "-T") and value is the value for the option (e.g. "50").
+        If the option is a flag, set the value to None.
+
+    Returns
+    -------
+    burrito.util.CommandLineAppResult
+        It contains opened file handlers of stdout, stderr, and the 3
+        output files, which can be accessed in a dict style with the
+        keys of "StdOut", "StdErr", "--tblout". The exit status
+        of the run can be similarly fetched with the key of "ExitStatus".
+    '''
+    app = CMScan(InputHandler='_input_as_paths', params=params)
+    app.Parameters['--incE'].on(evalue)
+    app.Parameters['--cpu'].on(cores)
+    app.Parameters['--tblout'].on(out_fp)
+    return app([cm, in_fp])
+
+
+def cmscan_sequence(cm, seq, evalue=0.01, cores=0, params=None):
+    '''scan a `skbio.Sequence` against a covariance model database.
+
+    Parameters
+    ----------
+    cm : str
+    seq : skbio.Sequence or its child classes for nucleotide sequence
+    cores : int
+        Number of CPU cores. Default to zero, i.e. running in serial-only mode.
+    evalue : float
+        Default to 0.01. Threshold E-value.
+    params : dict
+        Other command line parameters for cmscan. key is the option
+        (e.g. "-T") and value is the value for the option (e.g. "50").
+        If the option is a flag, set the value to None.
+
+    Returns
+    -------
+    Same data type as seq with added ncRNA annotation as positional_metadata.
```

Comment:
That's a nice idea. I should do that. It should take anything that can
convert to `Sequence` or its child object. I will add support for sequence
str too.

On Fri, Oct 16, 2015 at 9:52 AM, mortonjt <notifications@github.com> wrote:

> In micronota/bfillings/infernal.py
> <https://github.com/biocore/micronota/pull/21#discussion_r42263872>:
>
> > +    Parameters
> > +    ----------
> > +    cm : str
> > +    seq : skbio.Sequence or its child classes for nucleotide sequence
> > +    cores : int
> > +        Number of CPU cores. Default to zero, i.e. running in serial-only mode.
> > +    evalue : float
> > +        Default to 0.01. Threshold E-value.
> > +    params : dict
> > +        Other command line parameters for cmscan. key is the option
> > +        (e.g. "-T") and value is the value for the option (e.g. "50").
> > +        If the option is a flag, set the value to None.
> > +
> > +    Returns
> > +    -------
> > +    Same data type as seq with added ncRNA annotation as positional_metadata.
>
> What would you think about something like sequence_like?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/micronota/pull/21/files#r42263872>.
>